src/components/BrowserSelector.tsx

```diff
@@ -20,35 +20,30 @@ const browserIcons = {
 type BrowserData = Pick<BrowserInfo, "id" | "key" | "name">;
 
 interface BrowserSelectorProps {
+  selectedBrowser: BrowserData;
   browserResults: { browser: BrowserData; result: ComparisonResult }[];
   onSelectBrowser: (browser: BrowserData) => void;
 }
 
-export const BrowserSelector = ({ browserResults, onSelectBrowser }: BrowserSelectorProps) => {
-  const [selected, setSelected] = React.useState(browserResults[0].browser);
-
-  const handleSelect = React.useCallback(
-    (browser: BrowserData) => {
-      setSelected(browser);
-      onSelectBrowser(browser);
-    },
-    [onSelectBrowser]
-  );
-
+export const BrowserSelector = ({
+  selectedBrowser,
+  browserResults,
+  onSelectBrowser,
+}: BrowserSelectorProps) => {
   const links = browserResults
     .filter(({ browser }) => browser.key in browserIcons)
     .map(({ browser, result }) => ({
-      active: selected === browser,
+      active: selectedBrowser === browser,
       id: browser.id,
-      onClick: () => handleSelect(browser),
+      onClick: () => onSelectBrowser(browser),
       right: result !== ComparisonResult.Equal && <StatusDot status={result} />,
       title: browser.name,
     }));
```

Comment:
wrap in `useMemo`, perhaps?

Reply:
Hmm, I would be loathe to start adding `useMemo` before seeing a perf issue. Premature optimization and all that. I don't think this snippet is doing much?